docs/refs/crd-upgrade-safety.md

```diff
@@ -0,0 +1,210 @@
+# Custom Resource Definition Upgrade Safety
+
+OLM implements several validation checks when upgrading Custom Resource Definitions
+in order to maintain backwards compatibility with previous versions of that CRD.
+These validation checks are known collectively as the CRD Upgrade Safety preflight check.
+
+The CRD Upgrade Safety preflight check consists of several individual field validations
+that are performed before an upgrade to a CRD is allowed to progress on a cluster.
+
+## Disallowed CRD Upgrade Changes
+
+The following changes to an existing CRD will be caught by the CRD Upgrade
+Safety preflight check and prevent the upgrade:
+- The scope changes from Cluster to Namespace or from Namespace to Cluster
+- An existing stored version of the CRD is removed
+- A new required field is added to an existing version of the CRD
+- An existing field is removed from an existing version of the CRD
+- An existing field type is changed in an existing version of the CRD
+- A new default value is added to a field that did not previously have a default value
+- The default value of a field is changed
+- An existing default value of a field is removed
+- New enum restrictions are added to an existing field which did not previously have enum restrictions
+- Existing enum values from an existing field are removed
+- The minimum value of an existing field is increased in an existing version
+- The maximum value of an existing field is decreased in an existing version
+- Minimum or maximum field constraints are added to a field that did not previously have constraints
```

Comment:
That could be nice. I spent a few minutes trying to think of how exactly that would be structured and couldn't really come up with a good idea. Maybe columns for safe/unsafe changes or something?

Reply:
I have played around with it too, but I haven't been able to beat your list. I also haven't been able to figure out a good information architecture.


I do feel like like it would be nice to have this be a bit more scan-able as a reference. Maybe we will come up with something when we downstream this doc?